test/cmd/newapp.sh

```diff
@@ -167,4 +167,7 @@ os::cmd::expect_success 'oc new-app mongo -o json | python -m json.tool'
 # Ensure custom branch/ref works
 os::cmd::expect_success 'oc new-app https://github.com/openshift/ruby-hello-world#beta4'
 
+# Ensure the resulting BuildConfig doesn't have unexpected sources
+os::cmd::expect_success_and_not_text 'oc new-app https://github.com/openshift/ruby-hello-world -o=jsonpath="{.items[?(@.kind==\"BuildConfig\")].spec.source}"' 'dockerfile|binary'
```

Comment:
do we have a test case that we're confident in for --dockerfile?  I don't see any newapp.sh invocation that tests it and you(@rhcarvalho) have made me nervous about the reliability of our integration tests due to the mocking.